src/api/app/models/user.rb

```diff
@@ -183,11 +152,23 @@ def self.find_with_credentials(login, password)
     end
 
     user = find_by_login(login)
-    # If the user could be found and the passwords equal then return the user
-    if user && user.password_equals?(password)
-      user.mark_login!
 
-      return user
+    if user
+      # check if the users password was hashed with an old hashing algorithm
+      if user.password_old
+        if user.password_old_equals?(password)
```

Comment:
You can do that in one line
```
if user.password_old && user.password_old_equals?(password)
```

Reply:
Nope. Otherwise it would jump to `user.authenticate` if the user entered a wrong old password and this would raise an exception since `password_digest` is not a BCrypt string.

Reply:
Then move this to a dedicated function (e.g. overwrite the authenticate function as I suggested in my other comment)

Reply:
done. see: https://github.com/openSUSE/open-build-service/pull/3814/files#diff-b94b02ede7519628bd289bf9ca49bf55R305